packages/sdk-cli/src/models/localRelay/relayInfo.ts

```diff
@@ -0,0 +1,93 @@
+import { cwd } from 'process';
+import { join } from 'path';
+import { waitUntil } from 'async-wait-until';
+
+import { isInt, readJsonFile } from './util';
+import { RELAY_PKG_NAME, RELAY_PID_FILE_PATH } from './const';
+
+export type RelayInfo = { port: number; pid: number };
+
+export type ReadRelayInfoResult = RelayInfo | false;
```

Comment:
In dynamically typed languages like JS (which TypeScript is laying static types on top of) this is much less common. It happens in languages like C mostly because the two memory layouts for a type like `RelayInfo | false` are totally different but since JS handles it for you it's fine (and makes errors less likely, since you can't accidentally use invalid `pid` values for example in this case.

Reply:
Oh I've never thought of that. I think, as Liam said, the most common style in JS is:

```
const result = fn();
if (!result) { /* handle error */ }
```

Reply:
Agreed, the original approach seems more JS oriented.